share/lib/python/neuron/tests/utils/__init__.py

```diff
@@ -0,0 +1,87 @@
+"""
+Utilities for writing tests
+"""
+from contextlib import contextmanager
+
+
+@contextmanager
+def cvode_enabled(enabled):
+    """
+    Helper for tests to enable/disable CVode, leaving the setting as they found it.
+
+    For example:
+    from neuron.tests.utils import cvode_enabled
+    with cvode_enabled(True):
+        # cvode is enabled here
+        some_test_using_cvode()
+    # cvode is disabled again
+    """
+    from neuron import h
+
+    old_status = h.cvode_active()
+    h.cvode_active(enabled)
+    try:
+        yield None
+    finally:
+        h.cvode_active(old_status)
+
+
+@contextmanager
+def cvode_use_long_double(enabled):
+    """
+    Helper for tests to enable/disable long double support in CVode, leaving the setting as they found it.
+    """
+    from neuron import h
+
+    cv = h.CVode()
+    old_setting = cv.use_long_double()
+    cv.use_long_double(enabled)
+    if enabled != old_setting:
+        # FIXME find a better way of forcing an already-initialised CVode to take the new use_long_double setting into account
```

Comment:
I'm happy to try ```NetCvode.structure_change()``` in the ```cv.use_long_double(enabled)``` implementation.
It it works, that could eliminate this if statement.